### PR TITLE
Feature/implement client states - login response

### PR DESCRIPTION
**Pull request by mrvary**
_split into a few parts because a lot of changes already..._

Update des Clients, sodass er die neue LOGIN_RESPONSE message annimmt. Nebenher diese auf dem Server vorbereitet.

### CODE REVIEW

File: src/client/src/react/lib.test.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ describe("setsWon", () => {
     expect(setsPlayer2).toBe(1);
   });
 
-  test("player1 does not won with 11 to 10 -> pl1: 0, pl2: 0", () => {
+  test("player1 does not win with 11 to 10 -> pl1: 0, pl2: 0", () => {
```
Comment (Adentalis) [inline src/client/src/react/lib.test.js]: 11:10 is better :P

File: src/client/src/react/lib.test.js [hunk 2]
```diff
@@ -110,4 +110,21 @@ describe("setsWon", () => {
     expect(setsPlayer1).toBe(0);
     expect(setsPlayer2).toBe(0);
   });
+
+  test.todo("test with more than one set");
+});
+
+describe("test isMatchFinished()", () => {
```
Comment (Adentalis) [inline src/client/src/react/lib.test.js]: in matchmaker/match -> getMatchWinner(match) this function is implemented as well

File: src/client/src/react/lib.js [hunk 3]
```diff
@@ -16,7 +17,12 @@ function setsWonPlayer2(match) {
   return setsWon(match, "player2", "player1");
 }
 
+function isMatchFinished(match) {
```
Comment (Adentalis) [inline src/client/src/react/lib.js]: in matchmaker/match -> getMatchWinner(match) 
is this function already  as well 
fyi

File: src/client/src/react/lib.test.js (hunk 1 above)
Comment (mrvary) [reply src/client/src/react/lib.test.js]: will fix it in the next run 👍

**Pull request merged**